python/ray/_private/runtime_env/working_dir.py

```diff
@@ -344,13 +364,17 @@ def fetch_package(self, pkg_uri: str) -> int:
         else:
             raise NotImplementedError(f"Protocol {protocol} is not supported")
 
-        _logger.debug(f"Unpack {pkg_file} to {local_dir}")
+        logger.debug(f"Unpacking {pkg_file} to {local_dir}")
         with ZipFile(str(pkg_file), "r") as zip_ref:
             zip_ref.extractall(local_dir)
         pkg_file.unlink()
+
         return local_dir
 
-    def upload_runtime_env_package_if_needed(self, job_config: JobConfig):
+    def upload_runtime_env_package_if_needed(
+            self,
+            job_config: JobConfig,
+            logger: Optional[logging.Logger] = default_logger):
```

Comment:
Do we need to pass in a logger where `upload_runtime_env_package_if_needed` is called, or is it okay to use the default logger?  Right now it's just using the default logger.

Reply:
I think it's fine to use the default logger for this because it's happening on the client side. The main place where we want to use the passed-in logger is in the agent to separate out the log files.